File: frontend/src/api/llm.ts
import api from './client';
import { llmBase } from './unifiedSync';

export interface LLMProvider {
  provider: string;
  model: string;
  connected: boolean;
  latency: number;
  icon_color: string;
  available: boolean;
}

export interface LLMStatusResponse {
  active_provider: string;
  active_model: string;
  connected: boolean;
  latency: number;
  providers: LLMProvider[];
}

export interface ActiveProvider {
  provider: string;
  model: string;
}

export interface SetActiveProviderRequest {
  provider: string;
  model: string;
}

export interface LLMTestResult {
  provider: string;
  model: string;
  connected: boolean;
  latency: number;
}

export interface OllamaModelsResponse {
  models: string[];
}

export interface LLMModelCatalogItem {
  id: string;
  name: string;
  provider: string;
  provider_model_id: string;
  description?: string;
  is_system: boolean;
  supports_streaming: boolean;
  context_length: number;
  price_input_per_1k: number;
  price_output_per_1k: number;
  currency: string;
}

export interface LLMModelCatalogResponse {
  models: LLMModelCatalogItem[];
}

export const getLLMStatus = () => api.get<LLMStatusResponse>(`${llmBase()}/llm/status`).then((res) => res.data);

export const getModelCatalog = () =>
  api.get<LLMModelCatalogResponse>(`${llmBase()}/llm/models/catalog`).then((res) => res.data);

// Ollama 模型列表始终查本机（桌面端用自己的 Ollama，与云端无关）
export const getOllamaModels = () =>
  api.get<OllamaModelsResponse>('/api/v1/llm/providers/ollama/models').then((res) => res.data);

export const getActiveProvider = () => api.get<ActiveProvider>('/api/v1/llm/active').then((res) => res.data);

export const setActiveProvider = (provider: string, model: string) =>
  api.post<ActiveProvider>('/api/v1/llm/active', { provider, model }).then((res) => res.data);

export const testProvider = (provider: string) =>
  api.post<LLMTestResult>(`/api/v1/llm/providers/${provider}/test`).then((res) => res.data);

export interface SummarizeRequest {
  text: string;
  length?: 'short' | 'medium' | 'long';
  model?: string;
}

export interface SummarizeResponse {
  summary: string;
  original_length: number;
  summary_length: number;
  compression_ratio: number;
  model_used: string;
  cached: boolean;
}

export const summarizeText = (data: SummarizeRequest) =>
  api.post<SummarizeResponse>('/api/v1/llm/summarize', data).then((res) => res.data);

export interface ExtractTagsRequest {
  text: string;
  max_tags?: number;
  suggest_categories?: boolean;
  model?: string;
}

export interface ExtractTagsResponse {
  tags: string[];
  categories?: string[];
  model_used: string;
}

export const extractTags = (data: ExtractTagsRequest) =>
  api.post<ExtractTagsResponse>('/api/v1/llm/extract-tags', data).then((res) => res.data);

export interface CompleteRequest {
  prompt: string;
  system_prompt?: string;
  model?: string;
  task_type?: string;
}

export interface CompleteResponse {
  text: string;
  model_used: string;
}

export const completeText = (data: CompleteRequest) =>
  api.post<CompleteResponse>('/api/v1/llm/complete', data).then((res) => res.data);

File: backend/app/schemas/llm.py
"""Pydantic schemas for LLM endpoints"""

from pydantic import BaseModel, Field
from typing import Optional, List, Dict
from enum import Enum


class SummarizeLength(str, Enum):
    SHORT = "short"
    MEDIUM = "medium"
    LONG = "long"


class ChatRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=100000, description="User message")
    history: Optional[List[Dict[str, str]]] = Field(None, description="Conversation history")
    brain_side: str = Field("both", description="Brain side context")
    sensitivity: str = Field("low", description="Content sensitivity level")
    task_type: str = Field("chat", description="Task type")
    preferred_model: Optional[str] = Field(None, description="Override routing model")
    system_prompt: Optional[str] = Field(None, description="System prompt override")
    conversation_id: Optional[str] = Field(None, description="会话 ID（传入则本轮问答落库到该会话）")


class SummarizeRequest(BaseModel):
    text: str = Field(..., min_length=10, max_length=50000, description="Text to summarize")
    length: SummarizeLength = Field(SummarizeLength.MEDIUM, description="Summary length: short/medium/long")
    model: Optional[str] = Field(None, description="Override model for summarization")


class SummarizeResponse(BaseModel):
    summary: str
    original_length: int
    summary_length: int
    compression_ratio: float
    model_used: str
    cached: bool = False


class ExtractTagsRequest(BaseModel):
    text: str = Field(..., min_length=5, max_length=50000, description="Text to extract tags from")
    max_tags: int = Field(10, ge=3, le=20, description="Maximum number of tags")
    suggest_categories: bool = Field(False, description="Also suggest categories")
    model: Optional[str] = Field(None, description="Override model for tag extraction")


class ExtractTagsResponse(BaseModel):
    tags: List[str]
    categories: Optional[List[str]] = None
    model_used: str


class CompleteRequest(BaseModel):
    prompt: str = Field(..., min_length=1, max_length=50000, description="Prompt text")
    system_prompt: Optional[str] = Field(None, max_length=50000, description="System prompt")
    model: Optional[str] = Field(None, description="Override model")
    task_type: str = Field("chat", max_length=50, description="Task type")


class CompleteResponse(BaseModel):
    text: str
    model_used: str


class EmbedRequest(BaseModel):
    text: str = Field(..., min_length=1, max_length=10000, description="Text to embed")
    store: bool = Field(False, description="Whether to store in database")
    content_type: str = Field("query", description="Content type for storage")
    content_id: Optional[str] = Field(None, description="Associated content ID")
    model: Optional[str] = Field(None, description="Embedding model label override (defaults to configured Ollama embed model)")


class EmbedBatchRequest(BaseModel):
    texts: List[str] = Field(..., min_length=1, max_length=50, description="List of texts to embed")
    model: Optional[str] = Field(None, description="Embedding model label override (defaults to configured Ollama embed model)")


class EmbedResponse(BaseModel):
    embedding: List[float]
    dimensions: int
    model_used: str


class EmbedBatchResponse(BaseModel):
    embeddings: List[List[float]]
    dimensions: int
    model_used: str
    count: int


class RouteTestRequest(BaseModel):
    message: str = Field(..., min_length=1, description="Message to test routing")
    brain_side: str = "both"
    sensitivity: str = "low"
    task_type: str = "chat"


class RouteTestResponse(BaseModel):
    provider: str
    model: str
    reasoning: str
    features_detected: List[str]
    is_sensitive: bool = False


class ModelInfoResponse(BaseModel):
    name: str
    provider: str
    description: str
    available: bool
    features: List[str]
    context_window: int
    latency_hint: str
    icon_color: str


class OllamaModelsResponse(BaseModel):
    models: List[str]
